### PR TITLE
Library build

### PR DESCRIPTION
**Pull request by bgermann**
This introduces a shared library build for libmediainfo. It is a prerequisite for publishing platform wheels as requested in #59.

### CODE REVIEW

File: cmake.patch [hunk 1]
```diff
@@ -0,0 +1,16 @@
+diff --git a/MediaInfoLib/Project/CMake/CMakeLists.txt b/MediaInfoLib/Project/CMake/CMakeLists.txt
+index 895965c2..e9d11ecd 100644
+--- a/MediaInfoLib/Project/CMake/CMakeLists.txt
++++ b/MediaInfoLib/Project/CMake/CMakeLists.txt
+@@ -85,11 +85,6 @@ elseif(APPLE)
+   set(CMAKE_DEBUG_POSTFIX "_debug")
+ endif()
+ 
+-# use bundled tinyxml only if no system
```
Comment (sbraz) [inline cmake.patch]: You can probably do `-DCMAKE_DISABLE_FIND_PACKAGE_TinyXML` instead of patching CMake, right?

**Comment by sbraz** [comment]
Wow that's a lot to review, I'm going to add a few comments in the areas I'm most familiar with.

File: .gitmodules [hunk 2]
```diff
@@ -1,3 +1,12 @@
...
 	url = git@github.com:paltman/pymediainfo.git
+[submodule "MediaInfoLib"]
+	path = MediaInfoLib
+	url = https://github.com/MediaArea/MediaInfoLib.git
+[submodule "ZenLib"]
+	path = ZenLib
+	url = https://github.com/MediaArea/ZenLib.git
+[submodule "zlib"]
+	path = zlib
+	url = https://github.com/madler/zlib.git
```
Comment (sbraz) [inline .gitmodules]: Do we really want to build against the git master of all these projects? That doesn't sound very stable.

Comment (bgermann) [reply .gitmodules]: The projects are at their latest release each. This is not configured in .gitmodule but in the "submodule links".

File: pymediainfo/__init__.py [hunk 3]
```diff
@@ -140,20 +140,25 @@ def _parse_xml_data_into_dom(xml_data, encoding_errors="strict"):
     @staticmethod
     def _get_library(library_file=None):
         os_is_nt = os.name in ("nt", "dos", "os2", "ce")
-        if library_file is not None:
+        if library_file is None:
+            lib_dir = 'lib'
             if os_is_nt:
-                return WinDLL(library_file)
+                lib_dir = 'bin'
+                dll = 'mediainfo.dll'
```
Comment (sbraz) [inline pymediainfo/__init__.py]: Is that case-insensitive? Will it find `MediaInfo.dll`?

File: pymediainfo/__init__.py [hunk 4]
```diff
@@ -140,20 +140,25 @@ def _parse_xml_data_into_dom(xml_data, encoding_errors="strict"):
...
+            elif sys.platform == "darwin":
+                dll = 'libmediainfo.dylib'
             else:
-                return CDLL(library_file)
-        elif os_is_nt:
-            return windll.MediaInfo
-        elif sys.platform == "darwin":
-            try:
-                return CDLL("libmediainfo.0.dylib")
```
Comment (sbraz) [inline pymediainfo/__init__.py]: Why was this removed? Will `ctypes.util.find_library` work instead?

**Comment by sbraz** [comment]
Looks like the travis build is failing, I'll let you handle that then I'll take another look. How can we tackle this libzen issue? Can't we ship it as well? The Windows DLL probably contains it already, right?

**Comment by sbraz** [comment]
Are you on any IRC server or can you join Gitter (I'm sbraz on Freenode and Gitter) ? I think that it would make communication easier for us.

**Comment by bgermann** [comment]
In the current configuration libzen.so is shipped as well. So a workaround would be loading it beforehand. But that is just as ugly as patching CMakeList.txt... I did not find a proper way to set ctypes' lib search path (some say, adding to PATH does the job but I would not do that in a library).

**Comment by sbraz** [comment]
@bgermann I'm getting a notification for every commit so I'm going to temporarily mute this PR. Please mention me when the CI is fixed and I'll re-enable them.

**Comment by bgermann** [comment]
See https://github.com/scikit-build/scikit-build/issues/357 for the PyPy issues.

**Comment by bgermann** [comment]
Closing in favour of #61.
